### PR TITLE
Upgrade to RESTEasy 4.5 to support async IO

### LINKED ISSUE #5593: Reduce down to a single Vert.x instance
### Description
Having two instances of Vert.x running is a large source of bugs.

@FroMage has done some initial work.

### Analysis
_Links to analysis docs (E.g Design, Requirements, etc)_

### Tasks
- [ ] _Checklist of sub tasks goes here_

### PR DESCRIPTION
**Pull request by stuartwdouglas**
- Remove second vertx instance
- Microprofile REST client 1.4

Fixes #5593

### CODE REVIEW

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/vertx/JsonArrayWriter.java [hunk 1]
```diff
@@ -35,4 +38,14 @@ public void writeTo(JsonArray jsonArray, Class<?> type, Type genericType, Annota
...
     }
+
+    @Override
+    public CompletionStage<Void> asyncWriteTo(JsonArray jsonArray, Class<?> type, Type genericType, Annotation[] annotations,
+            MediaType mediaType,
+            MultivaluedMap<String, Object> httpHeaders,
+            AsyncOutputStream entityStream) {
+        CompletionStage<Void> a = entityStream.asyncWrite(jsonArray.toBuffer().getBytes());
+        CompletionStage<Void> b = entityStream.asyncFlush();
+        return a.thenCompose(v -> b);
```
Comment (gastaldi) [inline extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/vertx/JsonArrayWriter.java]: Can't `asyncFlush` happen before `asyncWrite` in this snippet? Wondering if it's better to read:

```suggestion
        return entityStream.asyncWrite(jsonArray.toBuffer().getBytes())
                       .thenCompose(v -> entityStream.asyncFlush());
```

**Review by gastaldi — APPROVED** [review]
LGTM

File: extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/vertx/JsonArrayWriter.java (hunk 1 above)
Comment (stuartwdouglas) [reply extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/vertx/JsonArrayWriter.java]: I just removed the flush, it looks like none of the other writers have it, and in general flushes are a bad idea as they can force chunked encoding when it is not nessesary.

**Comment by FroMage** [comment]
So let's merge?

**Comment by FroMage** [comment]
Thanks!

**Comment by gsmet** [comment]
Thanks to you for your efforts.

Going to work on wrapping CR1 now.

**Pull request merged**